sda/internal/database/db_functions.go

```diff
@@ -288,3 +290,65 @@ func (dbs *SDAdb) getArchived(user, filepath, checksum string) (string, int, err
 
 	return filePath, fileSize, nil
 }
+
+// CheckAccessionIdExists validates if an accessionID exists in the db
+func (dbs *SDAdb) CheckAccessionIDExists(accessionID string) (bool, error) {
+	var err error
+	var exists bool
+	// 2, 4, 8, 16, 32 seconds between each retry event.
+	for count := 1; count <= RetryTimes; count++ {
+		exists, err = dbs.checkAccessionIDExists(accessionID)
+		if err == nil {
+			break
+		}
+		time.Sleep(time.Duration(math.Pow(2, float64(count))) * time.Second)
+	}
+
+	return exists, err
+}
+func (dbs *SDAdb) checkAccessionIDExists(accessionID string) (bool, error) {
+	dbs.checkAndReconnectIfNeeded()
+	db := dbs.DB
+	const checkIDExist = "SELECT COUNT(*) FROM sda.files WHERE stable_id = $1;"
```

Comment:
out of curiosity, should the `select` on line 282 still be from `local_ega`?

Reply:
Of course it should not be.